packages/ui/components/form/wizard/WizardForm.tsx

```diff
@@ -30,48 +33,52 @@ function WizardForm<T extends DefaultStep>(props: {
   const setStep = (newStep: number) => {
     router.replace(`${href}?step=${newStep || 1}`, undefined, { shallow: true });
   };
+  const [currentStepIsLoading, setCurrentStepIsLoading] = useState(false);
+
+  useEffect(() => {
+    setCurrentStepIsLoading(false);
+  }, [currentStep]);
```

Comment:
This is to be able to delegate the `isLoading` state to the step itself if it includes the callable element for that purpose. With this I was able to remove all `useStates` you guys were noticing @zomars @sean-brydon.

Reply:
WDYM? I think I don't follow.

Reply:
Instead of having an external state for each step which the step itself needs to manipulate and the wizard needs to know the result, I just made the step manipulate its own loading state and allowing the wizard access to it.